src/model_api.jl

```diff
@@ -182,3 +184,56 @@ If for some reason a model is sometimes unable to report feature importances the
 
 """
 function feature_importances end
+
+_named_tuple(named_tuple::NamedTuple) = named_tuple
+_named_tuple(::Nothing) = NamedTuple()
+_named_tuple(something_else) = (report=something_else,)
+_scrub(x) = x
+_scrub(x::NamedTuple) = isempty(x) ? nothing : x
+_keys(named_tuple) = keys(named_tuple)
+_keys(::Nothing) = ()
+
+"""
+    MLJModelInterface.report(model, report_given_method)
+
+Merge the reports in the dictionary `report_given_method` into a single
+property-accessible object. It is supposed that each key of the dictionary is either
+`:fit` or the name of an operation, such as `:predict` or `:transform`. Each value will be
+the `report` component returned by a training method (`fit` or `update`) dispatched on the
+`model` type, in the case of `:fit`, or the report component returned by an operation that
+supports reporting.
+
+# New model implementations
+
+Overloading this method is optional, unless the model generates reports that are neither
+named tuples nor `nothing`.
+
+Assuming each value in the `report_given_method` dictionary is either a named tuple 
+or `nothing`, and there are no conflicts between the keys of the dictionary values 
+(the individual reports), the fallback returns the usual named tuple merge of the 
+dictionary values, ignoring any `nothing` value. If there is a key conflict, all operation 
```

Comment:
I'm not sure the users would understand what we mean by operations. Would you consider `fit` and operation. 
The reason I'm asking this is because for the case of key conflict, the `:fit` report isn't wrapped in a named tuple.

Reply:
Right, `:fit` is *not* an operation. I think "operation" is explained earlier, but to make sure I've added a comment that `:fit` reports are never wrapped.